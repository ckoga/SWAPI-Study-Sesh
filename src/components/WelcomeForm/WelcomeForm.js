import React, {Component} from 'react';
import './WelcomeForm.scss';

class WelcomeForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '"Luke. I am your father."',
      rank: 'Padawan',
      // would we need a state property for each option that would be a boolean?
      error: false,
    }
  }

  render() {
    return(
      <form className='form'>
        <label>Your Name</label>
        <input type='text' name='name' value={this.state.name} />
        <label>Your Favorite SW Quote</label>
        <input type='text' name='quote' value={this.state.quote} />
        <label>Your SW Knowledge  </label>

        <label className='radios' htmlFor='rank1'>
          <input type='radio' name='rank' value='Padawan'id='rank1'
            checked={this.state.rank === 'Padawan'}
            onChange={this.handleRankSelection}
          />Padawan
        </label>

        <label className='radios' htmlFor='rank2'>
          <input type='radio' name='rank' value='Jedi Knight'id='rank2'
            checked={this.state.rank === 'Jedi Knight'}
            onChange={this.handleRankSelection}
          />Jedi Knight
        </label>

        <label className='radios' htmlFor='rank3'>
          <input type='radio' name='rank' value='Jedi Master'id='rank3'
            checked={this.state.rank === 'Jedi Master'}
            onChange={this.handleRankSelection}
          />Jedi Master
        </label>

        <button>ENTER</button>
      </form>
    )
  }
}

export default WelcomeForm;
