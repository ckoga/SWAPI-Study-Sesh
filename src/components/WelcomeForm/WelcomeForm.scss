.form {
  color: #F3CA3E;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width:40%;
  text-align: left;
  margin: 50px auto 0 auto;
  font-size: 1.4em;
}

input,
button{
  margin: 15px 15px 15px 0;
  padding: 10px;
  font-size: 1em;
  width: 100
}

label {
    margin: 5px 0 5px 0;
}

.radios {
  margin-left: 40px;
}
