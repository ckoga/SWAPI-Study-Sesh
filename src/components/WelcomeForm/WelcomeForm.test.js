import React from 'react';
import {shallow} from 'enzyme';
import WelcomeForm from './WelcomeForm';

describe('WelcomeForm', () => {
  let mockUpdateUser;
  let wrapper;

   beforeEach(() => {
     mockUpdateUser = jest.fn();
     wrapper = shallow(<WelcomeForm
       updateUser={mockUpdateUser}
     />);
   })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is invoked', () => {
    const mockEvent = { target: {name: 'Robbie' , value: 'pants'}}
    const expected = 'pants';
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('Robbie')).toEqual(expected);
  });

  it('should update state when handleRankSelection is invoked', () => {
    const mockEvent = {target: {value: 'pants'}};
    const expected = 'pants';
    wrapper.instance().handleRankSelection(mockEvent);
    expect(wrapper.state('rank')).toEqual(expected);
  });

  // it('should update state when handleClick is invoked with inputs completed',
  //   () => {
  //   this has an if/else conditional so probably will need to test both
  // });
  //
  // it('should update state when handleClick is invoke with empty input(s)',
  //   () => {
  //
  // });

  it('should invoke handleChange when changes occur', () => {
    const mockEvent = { target: {name: 'Robbie' , value: 'pants'}}
    wrapper.instance().handleChange = jest.fn();
    wrapper.find('#nameInput').simulate('change', mockEvent)
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
  });

  it('should invoke handleRankSelection when selection changes', () => {
    const mockEvent = {target: {value: 'pants'}};
    wrapper.instance().handleRankSelection = jest.fn();
    wrapper.find('#rank3').simulate('change', mockEvent)
    expect(wrapper.instance().handleRankSelection).toHaveBeenCalledWith(mockEvent)
  });

  it('should invoke handleClick when button is clicked', () => {
    const mockState = {name: 'Robbie', quote: '"Luke. I am your father."',
      rank: 'Padawan', error: false,};
    wrapper.setState(mockState);
    expect(wrapper.state()).toEqual(mockState);

    wrapper.instance().handleClick = jest.fn();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().handleClick).toHaveBeenCalled();
  });

  // it('should invoke updateUser with user object when handleClick is invoked',
  //   () => {
  //
  // });


});
