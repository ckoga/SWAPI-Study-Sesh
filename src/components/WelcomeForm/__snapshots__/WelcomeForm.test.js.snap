// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeForm should match the snapshot with all data passed in correctly 1`] = `
<form
  className="form"
>
  <label>
    Your Name
  </label>
  <input
    id="nameInput"
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label>
    Your Favorite SW Quote
  </label>
  <input
    name="quote"
    onChange={[Function]}
    type="text"
    value="\\"Luke. I am your father.\\""
  />
  <label>
    Your SW Knowledge
  </label>
  <label
    className="radios"
    htmlFor="rank1"
  >
    <input
      checked={true}
      id="rank1"
      name="rank"
      onChange={[Function]}
      type="radio"
      value="Padawan"
    />
    Padawan
  </label>
  <label
    className="radios"
    htmlFor="rank2"
  >
    <input
      checked={false}
      id="rank2"
      name="rank"
      onChange={[Function]}
      type="radio"
      value="Jedi Knight"
    />
    Jedi Knight
  </label>
  <label
    className="radios"
    htmlFor="rank3"
  >
    <input
      checked={false}
      id="rank3"
      name="rank"
      onChange={[Function]}
      type="radio"
      value="Jedi Master"
    />
    Jedi Master
  </label>
  <p
    className="error"
    hidden={true}
  >
    Error - Please provide name, quote and ranking to proceed.
  </p>
</form>
`;
